File: apps/mobile/lib/data/recipe_presets.dart
// lib/data/recipe_presets.dart

/// 食谱表单中常用的预设值。后续可能改为运行时可配置；本期为静态常量。
class RecipePresets {
  RecipePresets._();

  /// 分类预设。"+ 其他" 由 wrapper widget 在末尾追加。
  static const List<String> categories = [
    '家常',
    '川菜',
    '粤菜',
    '西式',
    '烘焙',
    '汤羹',
  ];

  /// 烹饪时间预设（分钟）。最后一个 120 在 UI 上展示为 "120+"，但点击仍写值 120。
  static const List<int> cookingMinutes = [15, 30, 45, 60, 90, 120];

  /// 食材单位预设。"自定义…" 由 unit_dropdown 在 sheet 末尾追加。
  static const List<String> units = [
    'g',
    'ml',
    'kg',
    '个',
    '把',
    '根',
    '颗',
    '片',
    '杯',
    '勺',
    '适量',
  ];
}

File: apps/ios/FreshPantry/Domain/Rules/RecipePresets.swift
import Foundation

/// Static preset values surfaced in the custom-recipe authoring form. Ported from
/// `lib/data/recipe_presets.dart`. These are recipe-domain presets (cuisine
/// categories / cooking-time chips / ingredient units) and are deliberately
/// SEPARATE from `FoodCategories` (the inventory perishability taxonomy) — a
/// recipe's category is a freeform cuisine label, not one of the five inventory
/// buckets.
enum RecipePresets {
    /// Cuisine category presets. A custom value the user typed is appended ahead
    /// of these by the form; the trailing "其他" sentinel opens a freeform entry.
    static let categories = ["家常", "川菜", "粤菜", "西式", "烘焙", "汤羹"]

    /// Cooking-time presets (minutes). The last value (120) renders as "120+" but
    /// still writes 120 on tap.
    static let cookingMinutes = [15, 30, 45, 60, 90, 120]

    /// Ingredient unit presets. A "自定义…" entry is appended by the picker.
    static let units = ["g", "ml", "kg", "个", "把", "根", "颗", "片", "杯", "勺", "适量"]
}
